contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,234 @@
+        uint256 newSweptETH = address(this).balance - consensusRewards;
+        accountingValid = true;
+
+        // send the ETH that is from fully withdrawn validators to the Vault
+        if (newSweptETH >= MAX_STAKE) {
+            uint256 fullyWithdrawnValidators;
+            // safe since MAX_STAKE is hardcoded to 32ETH
+            unchecked {
+                // explicitly cast to uint256 as we want to round to a whole number of validators
+                fullyWithdrawnValidators = uint256(newSweptETH / MAX_STAKE);
+            }
+            activeDepositedValidators -= fullyWithdrawnValidators;
+
+            uint256 wethToVault = MAX_STAKE * fullyWithdrawnValidators;
+            IWETH9(WETH_TOKEN_ADDRESS).deposit{ value: wethToVault }();
+            IWETH9(WETH_TOKEN_ADDRESS).transfer(VAULT_ADDRESS, wethToVault);
+
+            emit AccountingFullyWithdrawnValidator(
+                fullyWithdrawnValidators,
+                activeDepositedValidators,
+                wethToVault
+            );
+        }
+
+        uint256 ethRemaining = address(this).balance - consensusRewards;
+        // should be less than a whole validator stake
+        require(ethRemaining < 32 ether, "unexpected accounting");
+
+        // If no Beacon chain consensus rewards swept
+        if (ethRemaining == 0) {
+            // do nothing
+            return accountingValid;
+        }
+        // Beacon chain consensus rewards swept (partial validator withdrawals)
+        else if (ethRemaining < fuseIntervalStart) {
+            // solhint-disable-next-line reentrancy
+            consensusRewards += ethRemaining;
+            emit AccountingConsensusRewards(ethRemaining);
+        }
+        // Beacon chain consensus rewards swept but also a slashed validator fully exited
+        else if (ethRemaining > fuseIntervalEnd) {
+            IWETH9(WETH_TOKEN_ADDRESS).deposit{ value: ethRemaining }();
+            IWETH9(WETH_TOKEN_ADDRESS).transfer(VAULT_ADDRESS, ethRemaining);
+            activeDepositedValidators -= 1;
+
+            emit AccountingValidatorSlashed(
+                activeDepositedValidators,
+                ethRemaining
+            );
+        }
+        // Oh no... Fuse is blown. The Strategist needs to adjust the accounting values.
+        else {
+            return _failAccounting(pauseOnFail);
+        }
+    }
+
+    /// @dev pause any further accounting if required and return false
+    function _failAccounting(bool pauseOnFail)
+        internal
+        returns (bool accountingValid)
+    {
+        // pause if not already
+        if (pauseOnFail) {
+            _pause();
+        }
+        // fail the accounting
+        accountingValid = false;
+    }
+
+    /// @notice Allow the Strategist to fix the accounting of this strategy and unpause.
+    /// @param _validatorsDelta adjust the active validators by plus one, minus one or unchanged with zero
```

Comment:
Comment here says +1, but code seems to be +3?

Reply:
yup thanks, fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2015/commits/fc48427a5acb235499913c7705af009a7b4a961c